python/testmonitor/results/results.py

```diff
@@ -0,0 +1,224 @@
+
+
+def measure_power(current: float, voltage: float = 0) -> object:
+    """
+    Simulates taking an electrical power measurement.
+    This introduces some random current and voltage loss.
+    :param current: The electrical current value.
+    :param voltage: The electrical voltage value.
+    :return: A tuple containing the electrical power measurements and the input and output lists.
+    """
+    current_loss = 1 - random.uniform(0, 1) * 0.25
+    voltage_loss = 1 - random.uniform(0, 1) * 0.25
+    power = current * current_loss * voltage * voltage_loss
+
+    # Record electrical current and voltage as inputs.
+    inputs = [
+        NamedValueObject(name="current", value=current),
+        NamedValueObject(name="voltage", value=voltage)
+    ]
+
+    # Record electrical power as an output.
+    outputs = [
+        NamedValueObject(name="power", value=power)
+    ]
+
+    return power, inputs, outputs
+
+
+def build_power_measurement_params(
+        power: float,
+        low_limit: float,
+        high_limit: float,
+        status: object) -> object:
+    """
+    Builds a Test Monitor measurement parameter object for the power test.
+    :param power: The electrical power measurement.
+    :param low_limit: The value of the low limit for the test.
+    :param high_limit: The value of the high limit for the test.
+    :param status: The measurement's pass/fail status.
+    :return: A list of test measurement parameters.
+    """
+    parameter = {
+        "name": "Power Test",
+        "status": str(status.status_type),
+        "measurement": str(power),
+        "units": "Watts",
+        "nominalValue": None,
+        "lowLimit": str(low_limit),
+        "highLimit": str(high_limit),
+        "comparisonType": "GELE"
+    }
+
+    parameters = {"text": "", "parameters": [parameter]}
+    return parameters
+
+
+def generate_step_data(
+        name,
+        step_type,
+        inputs=None,
+        outputs=None,
+        parameters=None,
+        status=None):
+    """
+    Creates a <see cref="StepData"/> object and
+    populates it to match the TestStand data model.
+    :param name: The test step's name.
+    :param step_type: The test step's type.
+    :param inputs: The test step's input values.
+    :param outputs: The test step's output values.
+    :param parameters: The measurement parameters.
+    :param status:
+    :return: The step data used to create a test step.
+    """
+    step_status = status if status is not None else StatusObject(status_type='RUNNING')
+
+    step_data = TestStepRequestObject(
+        step_id=None,
+        parent_id=None,
+        result_id=None,
+        children=None,
+        data=parameters,
+        data_model="TestStand",
+        name=name,
+        started_at=None,
+        status=step_status,
+        step_type=step_type,
+        total_time_in_seconds=random.uniform(0, 1) * 10,
+        inputs=inputs,
+        outputs=outputs)
+
+    return step_data
+
+
+def main():
```

Comment:
This is optional since it's a `main`, but all other functions in this module have a docstring.

Reply:
I thought about that, and had read this post: https://stackoverflow.com/questions/49782603/does-main-need-a-docstring which links to https://gist.github.com/dolph/39d8f70cab6afbac8c01.  Their justification was that main() should do what the module is designed to do.  So, in that case, the module documentation at the top is essentially the same thing that you'd put as documentation for main().  In the second link, they actually put main() as the first method in the file so it comes immediately after the module documentation.  Thoughts?

Reply:
That seems reasonable to me. I suppose any docstring on it would always be super generic like "Entry point for the script" which is pretty unhelpful anyway because the use of `main` is obvious.